src/Token/Builder.php

```diff
@@ -169,4 +170,29 @@ public function getToken(Signer $signer, Key $key): Plain
             new Signature($signature, $encodedSignature)
         );
     }
+
+    private function formatClaims(array $claims): array
+    {
+        if (isset($claims[RegisteredClaims::AUDIENCE]) && count($claims[RegisteredClaims::AUDIENCE]) === 1) {
+            $claims[RegisteredClaims::AUDIENCE] = $claims[RegisteredClaims::AUDIENCE][0];
+        }
+
+        foreach (RegisteredClaims::DATE_CLAIMS as $claim) {
+            if (!isset($claims[$claim])) {
+                continue;
+            }
+
+            $claims[$claim] = $this->convertDate($claims[$claim]);
+        }
+
+        return $claims;
+    }
+
+    private function convertDate(DateTimeImmutable $date)
```

Comment:
I was with the old behaviour in my mind: https://3v4l.org/1Q7hh
Then when we convert it back to a `DateTime` we would have `DateTimeImmutable::createFromFormat('U.u', '1487372450.0231')` instead of `DateTimeImmutable::createFromFormat('U.u', '1487372450.023126')`, what can cause some weird errors.